src/Sentry/Internal/DefaultSentryStructuredLogger.cs

```diff
@@ -0,0 +1,131 @@
+        {
+            return;
+        }
+
+        var timestamp = _clock.GetUtcNow();
+
+        if (!TryGetTraceId(_hub, _scopeManager, out var traceId))
+        {
+            _options.DiagnosticLogger?.LogWarning("TraceId not found");
+        }
+
+        _ = TryGetParentSpanId(_hub, _scopeManager, out var parentSpanId);
+
+        string message;
+        try
+        {
+            message = string.Format(CultureInfo.InvariantCulture, template, parameters ?? []);
+        }
+        catch (FormatException e)
+        {
+            _options.DiagnosticLogger?.LogError(e, "Template string does not match the provided argument. The Log will be dropped.");
+            return;
+        }
+
+        SentryLog log = new(timestamp, traceId, level, message)
+        {
+            Template = template,
+            Parameters = ImmutableArray.Create(parameters),
+            ParentSpanId = parentSpanId,
+        };
+
+        try
+        {
+            configureLog?.Invoke(log);
+        }
+        catch (Exception e)
+        {
+            _options.DiagnosticLogger?.LogError(e, "The configureLog callback threw an exception. The Log will be dropped.");
+            return;
+        }
+
+        log.SetAttributes(_options);
+
+        var configuredLog = log;
+        if (_options.Experimental.BeforeSendLogInternal is { } beforeSendLog)
+        {
+            try
+            {
+                configuredLog = beforeSendLog.Invoke(log);
+            }
+            catch (Exception e)
+            {
+                _options.DiagnosticLogger?.LogError(e, "The BeforeSendLog callback threw an exception. The Log will be dropped.");
+                return;
+            }
+        }
+
+        if (configuredLog is not null)
+        {
+            //TODO: enqueue in Batch-Processor / Background-Worker
+            // see https://github.com/getsentry/sentry-dotnet/issues/4132
+            _ = _hub.CaptureEnvelope(Envelope.FromLog(configuredLog));
+        }
+    }
+
+    private static bool TryGetTraceId(IHub hub, IInternalScopeManager? scopeManager, out SentryId traceId)
+    {
+        if (hub.GetSpan() is { } span)
+        {
+            traceId = span.TraceId;
+            return true;
+        }
+
+        if (scopeManager is not null)
+        {
+            var currentScope = scopeManager.GetCurrent().Key;
+            traceId = currentScope.PropagationContext.TraceId;
+            return true;
+        }
+
+        traceId = SentryId.Empty;
+        return false;
+    }
+
+    private static bool TryGetParentSpanId(IHub hub, IInternalScopeManager? scopeManager, out SpanId? parentSpanId)
+    {
+        if (hub.GetSpan() is { } span && span.ParentSpanId.HasValue)
+        {
+            parentSpanId = span.ParentSpanId;
+            return true;
+        }
+
+        if (scopeManager is not null)
+        {
+            var currentScope = scopeManager.GetCurrent().Key;
+            parentSpanId = currentScope.PropagationContext.ParentSpanId;
```

Comment:
This method may return true with an out value for `parentSpanId` of `null`, if the `PropagationContext` was created using this constructor:
https://github.com/getsentry/sentry-dotnet/blob/2e1fff191a83a50dd113de845a4689a2d40fd654/src/Sentry/SentryPropagationContext.cs#L36-L40